src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -762,8 +764,14 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
             String ucCredentialsId = uc.getCredentialsId();
             if (ucCredentialsId != null) {
                 String url = getParameterString(uc.getUrl(), environment);
-                List<StandardUsernameCredentials> urlCredentials = CredentialsProvider.lookupCredentials(StandardUsernameCredentials.class, project,
-                                        ACL.SYSTEM, URIRequirementBuilder.fromUri(url).build());
+                List<StandardUsernameCredentials> urlCredentials = CredentialsProvider.lookupCredentials(
+                        StandardUsernameCredentials.class,
+                        project,
+                        project instanceof Queue.Task
```

Comment:
@MarkEWaite oops... this is a fix of a separate issue that slipped in... I can move it out to a separate PR if you want. See https://github.com/jenkinsci/credentials-plugin/blob/master/docs/consumer.adoc#retrieve-the-secret-from-a-specific-credentials-instance for the documentation on retrieving credentials

Reply:
Thinking more: if we move it to a separate PR, the separate PR would need to be merged first because we need to replicate this logic in `GitSCMTelescope.build(Item,SCM scm,SCMRevision)`... which might be why I included it in this PR!

This change IIUC will fix https://issues.jenkins-ci.org/browse/JENKINS-44773 and https://issues.jenkins-ci.org/browse/JENKINS-44772 but not https://issues.jenkins-ci.org/browse/JENKINS-44774

Reply:
Hmmmm ffc9b9f should have fixed JENKINS-44774 though (unless that is something specialized)

Reply:
I don't see any significant benefit of moving it to a separate PR (at least not for me).

Is it worth the effort to make it a separate commit so that we have a place to refer to [JENKINS-44772](https://issues.jenkins-ci.org/browse/JENKINS-44772) and [JENKINS-44773](https://issues.jenkins-ci.org/browse/JENKINS-44773)?

Reply:
Ok so JENKINS-44774 sounds like a mis-understanding of how authorize-project-plugin works. not relevant here AIUI